src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1493,19 +1493,22 @@
 
 ;; Wrap the fact-type so that Clojure equality and hashcode semantics are used
 ;; even though this is placed in a Java map.
-(deftype AlphaRootsWrapper [roots roots-hash]
+(deftype AlphaRootsWrapper [fact-type fact-type-hash roots]
   Object
   (equals [this other]
     ;; Since we memoize the function that retrieves these on the fact type, and since the alpha roots for
     ;; any given type primary type T (excluding descendants and ancestors) are a disjoint
     ;; set with respect to the alpha roots for any other primary type T2, we know that we will
     ;; only have one AlphaRootsWrapper object for any given set of alpha roots per get-alphas-fn instance.
```

Comment:
This comment about having only one AlphaRootsWrapper instance isn't accurate (and never was, which is why this is currently not behaving the way https://github.com/cerner/clara-rules/issues/236 intended it to) and should be removed.